articles/techniques/operations-and-functions.md

```diff
@@ -125,22 +142,20 @@ function Square(x : Double) : (Double) {
 ```
 Whenever it is possible to do so, it is helpful to write out classical logic in terms of functions rather than operations, so that it can be more readily used from within operations.
 If we had written `Square` as an operation, for example, then the compiler would not have been able to guarantee that calling it with the same input would consistently produce the same outputs.
-This is especially critical when considering operation variants.
 
 To underscore the difference between functions and operations, consider the problem of classically sampling a random number from within a Q# operation:
 
 ```qsharp
 operation U(target : Qubit) : Unit {
-    body (...) {
-        let angle = RandomReal()
-        Rz(angle, target)
-    }
+
+    let angle = RandomReal()
+    Rz(angle, target)
 }
 ```
 
 Each time that `U` is called, it will have a different action on `target`.
-In particular, the compiler cannot guarantee that if we added an `Adjoint auto` statement to `U`, then `U(target); Adjoint U(target);` acts as identity (that is, as a no-op).
-This violates the definition of the adjoint that we saw in [Vectors and Matrices](xref:microsoft.quantum.concepts.vectors), such that allowing `Adjoint auto` in an operation where we have called the operation @"microsoft.quantum.canon.randomreal" would break the guarantees provided by the compiler; @"microsoft.quantum.canon.randomreal" is an operation for which no adjoint or controlled version exists.
+In particular, the compiler cannot guarantee that if we added an `adjoint auto` specialization declaration to `U`, then `U(target); Adjoint U(target);` acts as identity (that is, as a no-op).
+This violates the definition of the adjoint that we saw in [Vectors and Matrices](xref:microsoft.quantum.concepts.vectors), such that allowing to auto-generate an adjoint specialization in an operation where we have called the operation @"microsoft.quantum.canon.randomreal" would break the guarantees provided by the compiler; @"microsoft.quantum.canon.randomreal" is an operation for which no adjoint or controlled version exists.
```

Comment:
```suggestion
This violates the definition of the adjoint that we saw in [Vectors and Matrices](xref:microsoft.quantum.concepts.vectors), such that allowing to auto-generate an adjoint specialization in an operation where we have called the operation @"microsoft.quantum.math.randomreal" would break the guarantees provided by the compiler; @"microsoft.quantum.math.randomreal" is an operation for which no adjoint or controlled version exists.
```